### DISCUSSION TITLE
Argo CD support failure recovery?

### DISCUSSION THREAD

**Discussion by vinupatil2008** [argoproj/argo-cd#6229] [Q&A] [answered]
I have a scenario,
Multiple deployments are done per namespace and Argo CD is looking at individual repository and their deployments.
Helm charts are being used for deployments(multiple stages of sync, in a single deployment 5 stages- each stage upgrades kubernetes resources).
**Question :** If new configuration is updated in git for helm charts and some stage is failed via helm deployment.
then what are option available in Argo CD to handle such failures?

**Comment by jgwest** [comment, marked as answer]
Argo CD's primary function is to look at Git repositories (or Helm charts), generate manifests based on those repositories (this is known as the target state) and compares the live state of the cluster with the target state, updating live resources as needed in order to match that target state.

So in a failure situation, you will presumably want to return your the state of resources to a known good state:
- Revert the commit(s) that causes your kubernetes deployment to fail, and sync the applications, or,
- [Rollback](https://argoproj.github.io/argo-cd/user-guide/commands/argocd_app_rollback/) to a previous deployment that Argo CD performed 

Note that this only reverts the cluster resources, it does not revert changes made to persistent storage (for example, if you performed a database migration on a failed deployment, you will need to handle this, Argo CD is only responsible ensuring that Kubernetes resources are consistent.)

**Reply by vinupatil2008** [reply]
Thanks Jonathan.
Your reply helped me to clear out many doubts.

**Reply by hammadzz** [reply]
@jgwest so is the default behaviour for error handling do nothing? Example app is running well and synced, then I update a manifest, use cli to trigger sync, the sync fails. Is my application in the last known working state or now entered a failed stated?

I am not clear from the docs. Perhaps I just need to check docs on what the kubectl apply does when it fails.
